src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -260,6 +260,56 @@ public TypeSymbolWithAnnotations SetIsAnnotated(CSharpCompilation compilation)
         public TypeSymbolWithAnnotations AsNullableReferenceType() => _extensions.AsNullableReferenceType(this);
         public TypeSymbolWithAnnotations AsNotNullableReferenceType() => _extensions.AsNotNullableReferenceType(this);
 
+        public (bool, TypeSymbolWithAnnotations) TryMergeNullability(TypeSymbolWithAnnotations other, VarianceKind variance)
+        {
+            (bool isNullableSuccess, bool? isNullable) = TryMergeIsNullable(IsNullable, other.IsNullable, variance);
+            if (isNullableSuccess)
+            {
+                (bool typeSuccess, TypeSymbol type) = TypeSymbol.TryMergeNullability(other.TypeSymbol, variance);
+                if (typeSuccess)
+                {
+                    return (true, Create(type, isNullable, CustomModifiers));
+                }
+            }
+            return (false, default);
+        }
+
+        private static (bool, bool?) TryMergeIsNullable(bool? a, bool? b, VarianceKind variance)
```

Comment:
>(bool, bool?) [](start = 23, length = 13)

nit: consider naming output tuple

Reply:
Currently all callers of `TryMergeNullability` deconstruct the result, so there's no need for names.

---
In reply to: [223825789](https://github.com/dotnet/roslyn/pull/29727#discussion_r223825789) [](ancestors = 223825789)